src/api/messages/getMessages.js

```diff
@@ -45,27 +49,28 @@ type ServerApiResponseMessages = {|
 |};
 
 /** Exported for tests only. */
-export const migrateMessages = (messages: ServerMessage[], identity: Identity): Message[] =>
-  messages.map(message => {
-    const { reactions, avatar_url: rawAvatarUrl, ...restMessage } = message;
-
-    return {
-      ...restMessage,
-      avatar_url: AvatarURL.fromUserOrBotData({
-        rawAvatarUrl,
-        email: message.sender_email,
-        userId: message.sender_id,
-        realm: identity.realm,
-      }),
-      reactions: reactions.map(reaction => {
-        const { user, ...restReaction } = reaction;
-        return {
-          ...restReaction,
-          user_id: user.id,
-        };
-      }),
-    };
-  });
+export const migrateMessages = (
+  messages: $ReadOnlyArray<ServerMessage>,
+  identity: Identity,
+): Message[] =>
+  messages.map(<M: Message>(message: ServerMessageOf<M>): M => ({
+    // (This casting looks unnecessary, but seems to nudge Flow in the
+    // right direction.)
+    ...(message: ServerMessageOf<M>),
```

Comment:
Hmm, does it? If I remove this cast, it seems like everything works fine.

Perhaps it was needed in some previous draft version, but isn't now?

Reply:
> Perhaps it was needed in some previous draft version, but isn't now?

Huh, I guess so! Thanks for catching this!